### PR TITLE
feat: CP-9360 wallet selector rework

### PR DESCRIPTION
**Pull request by vvava**
## Description

Our old method how we stored the different Wallet Extension has been deprecated we could not detect a lot of them and made the whole thing messy.

## Changes

Now we are using [ EICP-6963](https://eips.ethereum.org/EIPS/eip-6963) to detect them and add them to the providers list. We have an updated UI as well.

## Testing

Install multiple extensions -> go to the playground and try to connect

## Screenshots:

https://github.com/user-attachments/assets/dc0a930d-93c4-4ea9-8e06-bf60802c9d02

## Checklist for the author

Tick each of them when done or if not applicable.

- [x] I've covered new/modified business logic with Jest test cases.
- [x] I've tested the changes myself before sending it to code review and QA.

### CODE REVIEW

**Review by meeh0w — COMMENTED** [review]
I like where this is going, please let me know when it's ready for review (lots of commented code / console.logs at the moment)👏

File: src/background/services/web3/models.ts [hunk 1]
```diff
@@ -4,4 +4,7 @@ export enum WalletExtensionType {
   UNKNOWN = 'UNKNOWN',
   RABBY = 'RABBY',
   COINBASE = 'COINBASE',
+  PHANTOM = 'PHANTOM',
```
Comment (meeh0w) [inline src/background/services/web3/models.ts:7]: I can't really make Phantom appear, I guess they don't announce themselves? 🤔

**Review by meeh0w — APPROVED** [review]
This is awesome! I'm not sure about the coin-flip animation on hover, but other than that I really like it 🎉

**Review by meeh0w — CHANGES_REQUESTED** [review]
Actually, let's remove the console.logs 😬

**Review by bferenc — APPROVED** [review]
Works great, however (as Michal mentioned) there are some console logs in the changes.

**Review by meeh0w — APPROVED** [review]

File: src/background/services/web3/models.ts (hunk 1 above)
Comment (vvava) [reply src/background/services/web3/models.ts:7]: I'm checking that when I get home later today.

**Review by bferenc — DISMISSED** [review]

File: src/background/services/web3/models.ts (hunk 1 above)
Comment (vvava) [reply src/background/services/web3/models.ts:7]: image
It seems it can work :D

**Review by meeh0w — APPROVED** [review]

**Review by bferenc — APPROVED** [review]

**Pull request merged**